pkg/querier/multi_tenant_querier_test.go

```diff
@@ -148,3 +151,185 @@ func newSampleIterator() iter.SampleIterator {
+				ingester.On("Series", mock.Anything, req, mock.Anything).Return(nil, errors.New("tst-err"))
+				store.On("GetSeries", mock.Anything, mock.Anything).Return(nil, nil)
+			},
+			run: func(t *testing.T, q *MultiTenantQuerier, req *logproto.SeriesRequest) {
+				ctx := user.InjectOrgID(context.Background(), "1|2")
+				_, err := q.Series(ctx, req)
+				require.Error(t, err)
+			},
+		},
+		{
+			desc: "store error",
+			req:  mockSeriesRequest([]string{`{a="1"}`}),
+			setup: func(store *storeMock, querier *queryClientMock, ingester *querierClientMock, limits validation.Limits, req *logproto.SeriesRequest) {
+				ingester.On("Series", mock.Anything, req, mock.Anything).Return(mockSeriesResponse([]map[string]string{
+					{"a": "1"},
+				}), nil)
+				store.On("GetSeries", mock.Anything, mock.Anything).Return(nil, context.DeadlineExceeded)
+			},
+			run: func(t *testing.T, q *MultiTenantQuerier, req *logproto.SeriesRequest) {
+				ctx := user.InjectOrgID(context.Background(), "1|2")
+				_, err := q.Series(ctx, req)
+				require.Error(t, err)
+			},
+		},
+		{
+			desc: "no matches",
+			req:  mockSeriesRequest([]string{`{a="1"}`}),
+			setup: func(store *storeMock, querier *queryClientMock, ingester *querierClientMock, limits validation.Limits, req *logproto.SeriesRequest) {
+				ingester.On("Series", mock.Anything, req, mock.Anything).Return(mockSeriesResponse(nil), nil)
+				store.On("GetSeries", mock.Anything, mock.Anything).Return(nil, nil)
+			},
+			run: func(t *testing.T, q *MultiTenantQuerier, req *logproto.SeriesRequest) {
+				ctx := user.InjectOrgID(context.Background(), "1|2")
+				resp, err := q.Series(ctx, req)
+				require.Nil(t, err)
+				require.Equal(t, &logproto.SeriesResponse{Series: make([]logproto.SeriesIdentifier, 0)}, resp)
+			},
+		},
+		{
+			desc: "returns series",
+			req:  mockSeriesRequest([]string{`{a="1"}`}),
+			setup: func(store *storeMock, querier *queryClientMock, ingester *querierClientMock, limits validation.Limits, req *logproto.SeriesRequest) {
+				ingester.On("Series", mock.Anything, req, mock.Anything).Return(mockSeriesResponse([]map[string]string{
+					{"a": "1", "b": "2"},
+					{"a": "1", "b": "3"},
+				}), nil)
+
+				store.On("GetSeries", mock.Anything, mock.Anything).Return([]logproto.SeriesIdentifier{
+					{Labels: map[string]string{"a": "1", "b": "4"}},
+					{Labels: map[string]string{"a": "1", "b": "5"}},
+				}, nil)
+			},
+			run: func(t *testing.T, q *MultiTenantQuerier, req *logproto.SeriesRequest) {
+				ctx := user.InjectOrgID(context.Background(), "1|2")
+				resp, err := q.Series(ctx, req)
+				require.Nil(t, err)
+				require.ElementsMatch(t, []logproto.SeriesIdentifier{
+					{Labels: map[string]string{"a": "1", "b": "2"}},
+					{Labels: map[string]string{"a": "1", "b": "3"}},
+					{Labels: map[string]string{"a": "1", "b": "4"}},
+					{Labels: map[string]string{"a": "1", "b": "5"}},
```

Comment:
So, as we've discussed, the set should include `"__tenant_id__": "1"` and `"__tenant_id__": "2"` for each series.